db/hints/host_filter.cc

```diff
@@ -33,7 +33,7 @@ bool host_filter::can_hint_for(const locator::topology& topo, gms::inet_address
     case enabled_kind::enabled_for_all:
         return true;
     case enabled_kind::enabled_selectively:
-        return _dcs.contains(topo.get_datacenter(ep));
+        return topo.has_endpoint(ep, locator::topology::pending::yes) && _dcs.contains(topo.get_datacenter(ep));
```

Comment:
aside: pending should be renamed to also_pending or similar, since pending::yes doesn't exclude non-pending endpoints.

Reply:
even better!